staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -111,6 +112,47 @@ func (mounter *Mounter) hasSystemd() bool {
 	return *mounter.withSystemd
 }
 
+// Do a bind mount including the needed remount for applying the bind opts.
+// If the remount fails and we are running in a user namespace
+// figure out if the source filesystem has the ro, nodev, noexec, nosuid,
+// noatime, relatime or nodiratime flag set and try another remount with the found flags.
+func (mounter *Mounter) bindMountSensitive(mounterPath string, mountCmd string, source string, target string, fstype string, bindOpts []string, bindRemountOpts []string, bindRemountOptsSensitive []string, mountFlags []string, systemdMountRequired bool) error {
+	err := mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if err != nil {
+		return err
+	}
+	err = mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindRemountOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if libcontaineruserns.RunningInUserNS() {
+		if err == nil {
+			return nil
+		}
+		// Check if the source has ro, nodev, noexec, nosuid, noatime, relatime,
+		// nodiratime flag...
```

Comment:
Can you please put all this code into something like `func getUserNSBindMountOptions(path string, statfs func(...))` and unit test it with a fake `Statfs` implementation?

Reply:
> Can you please put all this code into something like `func getUserNSBindMountOptions(path string, statfs func(...))` and unit test it with a fake `Statfs` implementation?

You mean the code in lines 129 till 149?

Reply:
yes

Reply:
Something like in the last commit (0e13fa2)?

Reply:
Yes, thanks!